### PR TITLE
Add a template on how to implement a label provider for a custom file type #45

### PR DESCRIPTION
**Pull request by JonasHelming**
[**Description**](#description)
This PR adds a new template to the generator, which is a blueprint on how to implement LabelProviders for custom file types in Theia. The example label provider registers for the ".my" file extension

[**How to test:**](#how-to-test)
1. clone the repository, and checkout the appropriate branch
2. perform `yarn`
3. perform `npm link` (creates local copy instead of getting from npm directly)
4. create a directory on your filesystem and `cd` into it
5. perform `yo theia-extension`
6. select the `LabelProvider` option

### CODE REVIEW

**Comment by JonasHelming** [comment]
1. => done
2. => we could, but we have two authors so in terms of CLA, I belive two commits are cleaner
3. => You can create a file with ".my" as an extension

File: templates/labelprovider/frontend-module.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { ContainerModule } from "inversify";
+import { <%= params.extensionPrefix %>LabelProviderContribution } from './<%= params.extensionPath %>-contribution';
+import '../../src/browser/style/example.css';
+
+export default new ContainerModule(bind => {
+    // label binding
+    <% if (params.example) { %>
+    bind(LabelProviderContribution).to(<%= params.extensionPrefix %>LabelProviderContribution);
+    <% } %>
+});
```
Comment (vince-fugnitto) [inline templates/labelprovider/frontend-module.ts]: [minor] please include a newline.

File: templates/labelprovider/style/example.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    background-size: 12px;
+    width: 15px;
+    height: 10px;
+}
+.light-plus .my-icon{
+    background-image: url('./baseline_code_black_18dp.png');
+}
+.dark-plus .my-icon{
+    background-image: url('./baseline_code_white_18dp.png');
+}
```
Comment (vince-fugnitto) [inline templates/labelprovider/style/example.css]: [minor] please add a newline.

File: test/generator-test.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                        '.vscode/launch.json'
+                    ]);
+
+                    done();
+                } catch (e) {
+                    done(e);
+                }
+            }, done);
+    });
+});
```
Comment (vince-fugnitto) [inline test/generator-test.js]: [minor] please add a newline.

File: test/generator-test.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
+var assert = require('yeoman-assert')
```
Comment (vince-fugnitto) [inline test/generator-test.js]: I think we should probably open an issue to support CI in the repository to run these tests :)

File: templates/labelprovider/frontend-module.ts (hunk 1 above)
Comment (sdirix) [reply templates/labelprovider/frontend-module.ts]: Done

File: templates/labelprovider/style/example.css (hunk 2 above)
Comment (sdirix) [reply templates/labelprovider/style/example.css]: Done

File: test/generator-test.js (hunk 3 above)
Comment (sdirix) [reply test/generator-test.js]: Done

File: test/generator-test.js (hunk 4 above)
Comment (JonasHelming) [reply test/generator-test.js]: Do you want us to open this issue?

Comment (JonasHelming) [reply test/generator-test.js]: https://github.com/theia-ide/generator-theia-extension/issues/47

Comment (vince-fugnitto) [reply test/generator-test.js]: > #47

I've added it for you :)

**Comment by vince-fugnitto** [comment]
Thank you for the recent updates, it looks much better.
I noticed a minor issue, is it something you want to handle in your extension?

<div align='center'>


Screen Shot 2019-11-22 at 8 50 49 AM


</div>

Do we want to display the `icon` when opening the editor as well?

**Comment by JonasHelming** [comment]
We think it is a limitation of the labelprovider customization to not show an icon there, we usually provide a custom editor extension (just registering the normal text editor) and customize the icon there to achieve that. But this is strictly speaking not a "LabelProvider" example anymore, but more a "custom text file type"
We could contribute the editor example as well maybe as a "custom text file type" example.
If you dislike having the label provider separatly (as it is now), we could also only do the second example.

**Comment by vince-fugnitto** [comment]
> We think it is a limitation of the labelprovider customization to not show an icon there, we usually provide a custom editor extension (just registering the normal text editor) and customize the icon there to achieve that. But this is strictly speaking not a "LabelProvider" example anymore, but more a "custom text file type"
> We could contribute the editor example as well maybe as a "custom text file type" example.
> If you dislike having the label provider separatly (as it is now), we could also only do the second example.

I'm fine with the changes, just wanted to make you aware :)

**Comment by vince-fugnitto** [comment]
I am unable to publish a new version unfortunately, would it be possible @marcdumais-work @akosyakov whenever you get a chance?

**Comment by marcdumais-work** [comment]
> I am unable to publish a new version unfortunately, would it be possible @marcdumais-work @akosyakov whenever you get a chance?

I'll publish the new version now

**Comment by marcdumais-work** [comment]
> I'll publish the new version now

Done: [v0.1.16](https://github.com/theia-ide/generator-theia-extension/releases/tag/v0.1.16)

**Comment by vince-fugnitto** [comment]
Thank you @marcdumais-work !

**Pull request merged**